sui_types/src/batch.rs

```diff
@@ -3,21 +3,34 @@
 
 use crate::base_types::{AuthorityName, TransactionDigest};
 use crate::crypto::{sha3_hash, AuthoritySignature, BcsSignable};
+use crate::messages::TransactionInfoResponse;
 use serde::{Deserialize, Serialize};
 
 pub type TxSequenceNumber = u64;
 
 /// Either a freshly sequenced transaction hash or a batch
-#[derive(Eq, PartialEq, Clone, Debug, Serialize, Deserialize)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
 pub enum UpdateItem {
-    Transaction((TxSequenceNumber, TransactionDigest)),
+    Transaction(
+        (
+            TxSequenceNumber,
+            TransactionDigest,
+            Option<Box<TransactionInfoResponse>>,
```

Comment:
1. Might want to use struct syntax here, will be easier to read and refer to fields
2. Can we avoid the Box here?  I don't see why it's necessary
3. Why not have separate enums, `Transaction` and then `TransactionWithDetails` etc.?  Then no need for option.